src/com/github/Kraken3/AFKPGC/BotDetector.java

```diff
@@ -129,32 +145,25 @@ public synchronized void doDetector() {
 				// TODO: See if inconsistencies might be thread-safeness related.
 				if (lastActivities.containsKey(playerUUID) && p != null) {
 					LastActivity la = entry.getValue();
-					la.loggedLocations.add(p.getLocation());
-					if (la.loggedLocations.size() >= maxLocations) {
-						if (la.loggedLocations.size() > maxLocations) {
-							la.loggedLocations.removeFirst();
-						}
-						// we tracking location even if on reprieve or immune, but that's it
-						if (!reprieve.containsKey(playerUUID) && !AFKPGC.immuneAccounts.contains(playerUUID)) {
+					if (!reprieve.containsKey(playerUUID) && !AFKPGC.immuneAccounts.contains(playerUUID)) {
+						if (la.loggedLocations.size()<maxLocations) {

```

Comment:
It'll be equals to, not less than. In this case, use equals. Double check the prior logic. This code only execute if we had maxLocations or more, never less. Since you moved the add step above, this code should only ever execute on size() == maxLocations.